pipelines/context.go

```diff
@@ -0,0 +1,60 @@
+package pipelines
+
+import (
+	"context"
+)
+
+type contextKey struct{}
+
+var activePipelineKey = contextKey{}
+
+// ContextWithPipeline returns a copy of the given context which includes the pipeline p.
+func ContextWithPipeline(ctx context.Context, p Pipeline) context.Context {
+	return context.WithValue(ctx, activePipelineKey, p)
+}
+
+// PipelineFromContext returns the pipeline contained in the given context.
+func PipelineFromContext(ctx context.Context) (p Pipeline, ok bool) {
+	if ctx == nil {
+		return Pipeline{}, false
+	}
+	v := ctx.Value(activePipelineKey)
+	if s, ok := v.(Pipeline); ok {
+		return s, true
+	}
+	return Pipeline{}, false
+}
+
+// SetCheckpointOnContext sets a checkpoint on the pipeline in the context.
+func SetCheckpointOnContext(ctx context.Context, edgeName string) (Pipeline, context.Context) {
```

Comment:
Do we need the `*OnContext` suffix? Also, why not call the point `name`? Does it have to be `edgeName`?

Reply:
I changed edgeName to `edge`, what do you think?